ambry-frontend/src/main/java/com/github/ambry/frontend/FrontendUtils.java

```diff
@@ -133,4 +142,33 @@ static RestRequestMetricsGroup getMetricsGroupForGet(FrontendMetrics frontendMet
     }
     return group;
   }
+
+  /**
+   * Refresh the OperationOrBlobId in request path with the specific latest version info.
+   * @param restRequest the {@link RestRequest}
+   * @param datasetVersionRecord the {@link DatasetVersionRecord} which include the version info.
+   * @throws RestServiceException
+   */
+  static void refreshRequestPathWithNewOperationOrBlobIdIfNeeded(RestRequest restRequest,
```

Comment:
nit: rename this to replaceRequestPath... Refresh doesn't really make sense here.